### PR TITLE
feat: components add filter by version

### PR DESCRIPTION
**Pull request by xzdry**
https://user-images.githubusercontent.com/96771399/170400740-bb559d28-b7d0-47c7-a908-f997fbc9ce41.mov

### CODE REVIEW

File: packages/editor/src/components/ComponentsList/ComponentList.tsx [hunk 1]
```diff
@@ -40,6 +53,53 @@ function getCategoryOrder(name: string): number {
     : 0;
 }
 
+const FilterIcon = createIcon({
```
Comment (tanbowensg) [inline packages/editor/src/components/ComponentsList/ComponentList.tsx:56]: Why not use a chakra icon?

File: packages/editor/src/components/ComponentsList/ComponentList.tsx [hunk 2]
```diff
@@ -80,15 +156,74 @@ export const ComponentList: React.FC<Props> = ({ services }) => {
...
-        onChange={evt => setFilterText(evt.currentTarget.value)}
-      />
+      <InputGroup>
+        <Input
+          placeholder="filter the components"
+          value={filterText}
+          onChange={evt => setFilterText(evt.currentTarget.value)}
+        />
+        <InputRightElement>
+          <Popover closeOnBlur placement="bottom">
```
Comment (tanbowensg) [inline packages/editor/src/components/ComponentsList/ComponentList.tsx:170]: The popover should be wrappered to a new component. It is too chaos to mix with `ComponentList`

File: packages/editor/src/components/ComponentsList/ComponentList.tsx (hunk 1 above)
Comment (xzdry) [reply packages/editor/src/components/ComponentsList/ComponentList.tsx:56]: The chakra-icon does not appear to have a filter icon

File: packages/editor/src/components/ComponentsList/ComponentList.tsx (hunk 2 above)
Comment (xzdry) [reply packages/editor/src/components/ComponentsList/ComponentList.tsx]: done

**Review by tanbowensg — APPROVED** [review]

**Pull request merged**